app/experimenter/static/rapid/components/forms/ExperimentForm.tsx

```diff
@@ -40,38 +43,18 @@ const ExperimentForm: React.FC = () => {
 
   const handleSelectChange = (name) => {
     return (value) => {
-      dispatch({
-        type: ExperimentReducerActionType.UPDATE_STATE,
-        state: {
-          ...formData,
-          [name]: value,
-        },
-      });
+      dispatch(updateExperiment(name, value));
     };
   };
 
   const handleChange = (ev) => {
     const field = ev.target;
-    dispatch({
-      type: ExperimentReducerActionType.UPDATE_STATE,
-      state: {
-        ...formData,
-        [field.getAttribute("name")]: field.value,
-      },
-    });
+    dispatch(updateExperiment(field.getAttribute("name"), field.value));
   };
 
   const handleClickSave = async () => {
-    const url = experimentSlug
-      ? `/api/v3/experiments/${experimentSlug}/`
-      : "/api/v3/experiments/";
-    const response = await fetch(url, {
-      method: experimentSlug ? "PUT" : "POST",
-      headers: {
-        "Content-Type": "application/json",
-      },
-      body: JSON.stringify(formData),
-    });
+    const response = await saveExperiment(experimentSlug, formData);
+
```

Comment:
So you opted to just return the Response and unpack the errors externally?  It's fine we're gonna move that into its own reducer and we can just do it when we do that.